### ISSUE TITLE
Use GitHub secret for SonarCloud token

### ISSUE THREAD

**Issue by Marcono1234** [junit-pioneer/junit-pioneer#581]
It appears currently the GitHub workflow has the SonarCloud token hardcoded. This might allow malicious misuse of that token.
It would probably be better to use a [GitHub secret](https://docs.github.com/en/actions/security-guides/encrypted-secrets), similar to how it was done for the other secrets.

**Comment by beatngu13** [comment]
Please follow that thread:

https://twitter.com/beatngu1101/status/1407008608118771717

Do you have another idea? Otherwise, I would close this issue.

**Comment by Marcono1234** [comment]
I am not very familiar with SonarCloud, but to me it sounds that these tokens should be kept private because they can be used for more than just to "ruin our statistics" (which was mentioned in the Twitter thread):

- > We cannot recommend adding secrets to version control.

    [Source](https://community.sonarsource.com/t/how-to-use-sonarcloud-with-a-forked-repository-on-github/7363/10)
    
- > Why not just use an auth token? Because currently auth-tokens have all privileges of their source-users, so we really don’t want to put them out there in the wild.

    [Source](https://community.sonarsource.com/t/badges-on-private-projects/4894/2)
    
Though I am not sure if these statements are still up to date.
Also related:

- [SCCOMM-11](https://jira.sonarsource.com/browse/SCCOMM-11): Users should be able to create analysis token
- [Product Roadmap entry for analyzing external Pull Requests](https://portal.productboard.com/sonarsource/1-sonarcloud/c/50-sonarcloud-analyzes-external-pull-request) (you can also vote there for this feature)

**Comment by beatngu13** [comment]
Yes, the token _shouldn't_ be public. We already voted for the analysis of external PRs, but until then we have three options:

1. Skip SonarCloud analysis
2. Failing SonarCloud analysis (token hidden)
3. Working SonarCloud analysis (token exposed)

As we want to give contributors the possibility to analyize their PRs, we decided to go with option 3.

Looking at the [SonarCloud Web API](https://sonarcloud.io/web_api/), what is the worst that could happen? A compromised SonarCloud account and/or ruined statistics. We aware that this is far from ideal, but we are willing to make that tradeoff.

But please feel free to teach us better, as @nipafx said: "We're no security experts".

**Comment by Marcono1234** [comment]
To me this sounds like you have carefully considered the consequences, so I don't mind if you want to close this issue since there seems to be no better alternative at the moment.

**Comment by Michael1993** [comment]
This ticket is now indefinitely on hold, contingent on sonar implementing the new feature (analyze external PRs).

**Comment by nipafx** [comment]
Reading up on this I can confirm that Sonar still doesn't support this feature. There are (at least) two threads covering that topic:

* [Code analysis on pull request from forked repository with GitHub Actions](https://community.sonarsource.com/t/code-analysis-on-pull-request-from-forked-repository-with-github-actions/43986)
* [How to use SonarCloud with a forked repository on GitHub?](https://community.sonarsource.com/t/how-to-use-sonarcloud-with-a-forked-repository-on-github/7363)

In those threads, I found links to this issue that I can't see without logging and I don't think I have an account:

* [MMF-1371](https://jira.sonarsource.com/browse/MMF-1371)

I'll mark this issue as blocked and remove it from the milestone.

**Comment by nipafx** [comment]
'tis already marked as blocked 😊

**Comment by beatngu13** [comment]
After login:

<img width="454" alt="grafik" src="https://user-images.githubusercontent.com/11643714/169275752-9fb1619e-0ae7-4ace-b1a5-3760c4671464.png">

Nice… 😅

**Comment by beatngu13** [comment]
Via [SonarCloud's product roadmap](https://portal.productboard.com/sonarsource/1-sonarcloud/c/50):

>SonarCloud is now able to scan External Pull Requests when the project is configured to be scanned with Automatic Analysis ([https://docs.sonarsource.com/sonarcloud/advanced-setup/automatic-analysis/).](https://docs.sonarsource.com/sonarcloud/advanced-setup/automatic-analysis/)
>If your project uses Automatic Analysis, you do not need to activate anything to benefit from this feature; it should work out of the box.

However, at the moment we are using CI-based analysis with GitHub Actions, and the [SonarCloud documentation](https://docs.sonarsource.com/sonarcloud/advanced-setup/automatic-analysis/#conflict-with-ci-based-analysis) says:

>Automatic analysis is not intended to be used in conjunction with CI-based analysis.

We might want to switch then?
